Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -5421,35 +5421,47 @@ LayoutUnit RenderBox::computeLogicalWidthFromAspectRatio(RenderFragmentContainer
     return constrainLogicalWidthInFragmentByMinMax(logicalWidth, containerWidthInInlineDirection, *containingBlock(), fragment, AllowIntrinsic::No);
 }
 
+bool RenderBox::isReplacedElementWithIntrinsicRatio() const
+{
+    if (auto* replaced = dynamicDowncast<RenderReplaced>(this))
+        return replaced->computeIntrinsicAspectRatio();
```

Comment:
That's one of the main issues that I have been trying to figure out. Since not all Replaced Elements have an intrinsic ratio, right now they are being computed on demand. 

I think ideally we would store both the intrinsic and ratio as member fields. Then we could have a `RenderBox::hasAspectRatio` method that would check both `style().hasAspectRatio()` and the replaced element fields when needed.

Reply:
I wonder can we just use `replaced->intrinsicSize().isEmpty()` here?

Reply:
Unfortunately, I don't think that will work for all cases :( For example, a SVG could have a dimension missing but still have an intrinsic ratio. imported/w3c/web-platform-tests/css/css-flexbox/flex-aspect-ratio-img-row-010.html is an example of a test with such behavior